### ISSUE TITLE
AI Editor: Different units for Raids 

### ISSUE THREAD

**Issue by Halbarad1** [Sh0wdown/UnofficialCrusaderPatch#393]
Hi,
I was changing the Sheriffs AIC and wanted to give him different Raid Units for seperate raids: Slaves, Maceman, Assassins. 
So when I was changing the parameters RaidUnit1, RaidUnit2, RaidUnit3 and he did send out these kind of troops, but all of them together. That wasn't that good, since Slaves burned the other units when attacking, Assassins slowed them down, Maceman and Slaves were showing the enemy where the raid units were going so Assassins invisibility wasn't that useful. 
So is there any way to let him send all of these units out but let them attack seperatly (one Raid with Maceman, another one with Slaves, another one with Assassins)? If not it would be nice if it could be possible to implement such a strategy.

**Comment by Admiral-Gumpert** [comment]
I was trying to do the same thing , At the start of a game AIVs might actually seperate them like The Rat 
I made him Send out spearmen and archers for raids , At first he will Send them individually , but later he will send them together , and sometimes he will stop raiding which becomes annoying and making him even more useless

**Comment by Panbutt** [comment]
Yeah it definitely seems ... random. I've experimented with this myself by having an AI use slaves and knights for raiding. Sometimes they would send each group separately, other times they would be combined.  I'm wondering if it has something to do with the RecruitRaid rate and the RaidMax number.

**Comment by Heroesflorian** [comment]
Afaik, this also has to do with how the AI recruits raid troops (similar for other troops). 
They go through the list of raid units, top to bottom, and try to recruit the first one... if they can't (missing gold, equipment, etc), they pick the second one... and so on. When low on gold/horses/weapons, they'd most likely recruit slaves only as they cannot afford knights.

**Comment by ByBurton** [comment]
@Heroesflorian They don't just recruit from top to bottom. If they would do that, they would only buy the first ones if they are slaves for example, the least expensive unit in the game.
But if you give the caliph these raiding units (from top to bottom): Slave, ArabArcher you would probably see at least 75% more slaves if it would be like you said.
But the caliph will recruit almost 50 : 50. 
At least that is what happened on my tests.
